.github/workflows/integration-tests-reusable.yml

```diff
@@ -0,0 +1,108 @@
+name: Integration Tests (Reusable Workflow)
+
+on:
+  workflow_call:
+    inputs:
+      num_shards:
+        required: true
+        type: number
+
+env:
+  dotnet-version: 6.0.x # SDK Version for building Dafny
+  
+jobs:
+  # This workflow is used to dynamically calculate the matrix dimensions.
+  # For now that just makes the sharding a little cleaner, but it will shortly
+  # be used to switch between one platform and all platforms based on an input parameter.
+  populate-matrix-dimensions:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Populate OS list
+        id: populate-os-list
+        run: |
+          echo "::set-output name=os-list::[\"ubuntu-latest\", \"ubuntu-18.04\", \"macos-latest\", \"windows-latest\"]"
+      - name: Populate OS mapping for package.py
+        id: populate-os-mapping
+        run: |
+          echo "::set-output name=os-mapping::{\"ubuntu-latest\": \"ubuntu\", \"ubuntu-18.04\": \"ubuntu\", \"macos-latest\": \"osx\", \"windows-latest\": \"win\"}"
+      - name: Populate shard list
+        id: populate-shard-list
+        run: |
+          echo "::set-output name=shard-list::[" $(seq -s , 1 ${{ inputs.num_shards }}) "]"
+    outputs:
+      os-list: ${{ steps.populate-os-list.outputs.os-list }}
+      os-mapping: ${{ steps.populate-os-mapping.outputs.os-mapping }}
+      shard-list: ${{ steps.populate-shard-list.outputs.shard-list }}
+  test:
+    needs: populate-matrix-dimensions
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ${{ fromJson(needs.populate-matrix-dimensions.outputs.os-list) }}
+        shard: ${{ fromJson(needs.populate-matrix-dimensions.outputs.shard-list) }}
+      fail-fast: false
+    steps:
+    - name: Manually sync certs
+      if: runner.os == 'Linux'
+      run: cert-sync /etc/ssl/certs/ca-certificates.crt
+    - uses: actions/setup-python@v1
+    - name: Setup dotnet
+      uses: actions/setup-dotnet@v1.9.0
+      with:
+        dotnet-version: ${{env.dotnet-version}}
```

Comment:
Tiny nit, but the rest of the code has spaces after `{{` and before `}}`.

Reply:
That's fair, I'm going to batch that up with the next PR so I don't have to wait on the CI again if that's already.

Reply:
Yeah, not important at all!